File: src/atr_pkg/scripts/ma_node.py
#!/usr/bin/env python3
#coding=utf-8
import rospy
from std_msgs.msg import String
from qq_msgs.msg import Carry
def chao_cb(msg):
    rospy.loginfo(msg.data)
    
def yao_cb(msg):
    rospy.logwarn(msg.data)
    
def main():
    rospy.init_node("ma_node_py")
    sub_py = rospy.Subscriber("kaiheiqunpy",Carry,chao_cb)
    # sub_cpp = rospy.Subscriber("kaiheiqun",String,yao_cb)
    rospy.spin()
if __name__ == '__main__':
    main()
    

File: src/atr_pkg/src/ma_node.cpp
#include<ros/ros.h>
#include<std_msgs/String.h>
#include<qq_msgs/Carry.h>
void chao_cb(qq_msgs::CarryConstPtr msg)
{
    
    ROS_INFO(msg->data.c_str());
}
void yao_cb(qq_msgs::CarryConstPtr msg)
{
    ROS_WARN(msg->data.c_str());
}
int main(int argc, char *argv[])
{
    setlocale(LC_ALL,"");
    ros::init(argc,argv,"ma_node");
    ros::NodeHandle nh;
    ros::Subscriber sub_chao = nh.subscribe("kaiheiqun",1,chao_cb); 
    ros::Subscriber sub_yao = nh.subscribe("daifeiqun",1,yao_cb); 
    ros::spin();
    return 0;
}
